### PR TITLE
GEODE-9162: Mark Radish as experimental

### PR DESCRIPTION
**Pull request by jhutchison**
- make redis properties in ConfigurationProperties experimental
 - add experimental annotation to cli strings
 - add experimental marker to help messages

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/i18n/CliStrings.java [hunk 1]
```diff
@@ -2624,14 +2625,22 @@
   public static final String START_SERVER__PROPERTIES = "properties-file";
   public static final String START_SERVER__PROPERTIES__HELP =
       "The gemfire.properties file for configuring the Cache Server's distributed system. The file's path can be absolute or relative to the gfsh working directory.";
+  @Experimental
   public static final String START_SERVER__REDIS_BIND_ADDRESS =
       ConfigurationProperties.REDIS_BIND_ADDRESS;
+  @Experimental
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/management/internal/i18n/CliStrings.java:2631]: do we need to add `@experimental` tag to the `internal` packages?

File: geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java [hunk 2]
```diff
@@ -1965,7 +1980,12 @@
    * <U>Default</U>: 6379
    * </p>
    * <U>Allowed values</U>: 0..65535
+   *
+   * <p>
+   * Experimental: Geode API compatible with Redis is a new addition to Geode and the API may change
+   * <p/>
    */
+
   String REDIS_PORT = "compatible-with-redis-port";
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java:1989]: Here we will need the `@Experimental` tag

**Review by jinmeiliao — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/management/internal/i18n/CliStrings.java (hunk 1 above)
Comment (jhutchison) [reply geode-core/src/main/java/org/apache/geode/management/internal/i18n/CliStrings.java:2631]: I guess we don't -  I will remove.

**Review by Bill — CHANGES_REQUESTED** [review]
As @jinmeiliao pointed out: I believe you want the `@Experimental` annotation on each of these experimental fields.

**Review by mhansonp — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/management/internal/i18n/CliStrings.java (hunk 1 above)
Comment (jhutchison) [reply geode-core/src/main/java/org/apache/geode/management/internal/i18n/CliStrings.java]: done

File: geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java (hunk 2 above)
Comment (jhutchison) [reply geode-core/src/main/java/org/apache/geode/distributed/ConfigurationProperties.java:1991]: added-  thx

**Comment by jhutchison** [comment]
Requested changes made.  I may be being too persnickity here, but  does anyone know if the strings in the japicmp_exceptions are reviewed by docs?  If so can someone add @davebarnes97 as a reviewer here (I don't have permissions)

**Comment by davebarnes97** [comment]
Thanks for requesting a docs review, @jhutchison.

I would recommend against describing this feature as a "new addition."
What we're calling out here is that the	API might change. "Newness" as an explanation	will soon become stale,
but "Experimental" will	hold true over time.

Suggested re-wordings -- you may recognize the language from your contributions to the user guide:

In the several identical cases that begin with the tag "Experimental:", I suggest changing

"Experimental: Geode API compatible with Redis is a new addition to Geode and the API may change"

to

"Experimental: Geode API compatible with Redis is subject to change in future releases"

For the case in japicmp_exceptions.json, where the word "experimental" is not used, I would introduce it, replacing

"Geode API compatible with Redis is a new addition to Geode and the API may change"

with

"Geode API compatible with Redis is experimental and is subject to change in future releases"

**Review by davebarnes97 — APPROVED** [review]

**Review by Bill — APPROVED** [review]
Approved!

**Review by jinmeiliao — APPROVED** [review]

**Pull request merged**